Week9 - homework SQL/LibrariesManagement.sql

```diff
@@ -0,0 +1,298 @@
+
+INSERT INTO Books_Libraries
+VALUES(1, 1)
+INSERT INTO Books_Libraries
+VALUES(1, 3)
+INSERT INTO Books_Libraries
+VALUES(2, 1)
+INSERT INTO Books_Libraries
+VALUES(2, 3)
+INSERT INTO Books_Libraries
+VALUES(3, 3)
+INSERT INTO Books_Libraries
+VALUES(4, 1)
+INSERT INTO Books_Libraries
+VALUES(5, 4)
+INSERT INTO Books_Libraries
+VALUES(6, 1)
+INSERT INTO Books_Libraries
+VALUES(6, 3)
+INSERT INTO Books_Libraries
+VALUES(7, 4)
+INSERT INTO Books_Libraries
+VALUES(8, 1)
+INSERT INTO Books_Libraries
+VALUES(8, 3)
+
+SELECT *
+FROM Books_Libraries;
+SELECT *
+FROM Libraries;
+SELECT *
+FROM Books;
+
+INSERT INTO Sales
+VALUES(114, 1)
+INSERT INTO Sales
+VALUES(98, 4)
+INSERT INTO Sales
+VALUES(321, 3)
+INSERT INTO Sales
+VALUES(987, 2)
+INSERT INTO Sales
+VALUES(89, 7)
+INSERT INTO Sales
+VALUES(56, 8)
+INSERT INTO Sales
+VALUES(347, 5)
+INSERT INTO Sales
+VALUES(563, 11)
+INSERT INTO Sales
+VALUES(97, 6)
+INSERT INTO Sales
+VALUES(224, 9)
+INSERT INTO Sales
+VALUES(90, 12)
+INSERT INTO Sales
+VALUES(542, 10)
+
+SELECT *
+FROM Sales;
+
+--exercise2
+--Display all the librarians together with the library name they work in ordered by librarian name and library name descending.
+SELECT L.Name, Lib.Name
+FROM Librarians L
+INNER JOIN Libraries Lib ON Lib.Id = L.LibraryId
+ORDER BY L.Name DESC;
+
+SELECT L.Name, Lib.Name
+FROM Librarians L
+INNER JOIN Libraries Lib ON Lib.Id = L.LibraryId
+ORDER BY Lib.Name DESC;
+
+--exercise3
+--Update all the street names to start with "STR.".
+UPDATE Libraries
+SET StreetName = 'STR. ' + Streetname;   --I didn't know how to undo the 1st update. That's why the 1st try of 'str' is still there
+
+--exercise4
+--Display the authors' email for those who have more than 3 books published.
+SELECT AuthorEmail
+FROM Books
+GROUP BY AuthorEmail
+HAVING COUNT(AuthorLastName) > 3;
+
+--exercise5
+--Display the library name and the book title of the largest sale (the highest number of copies sold in one sale).
+SELECT MAX(S.NumberOfCopies), Lib.Name, B.Title    --I'm so stuck...
+FROM Sales S
+JOIN Books_Libraries BL ON S.BookLibraryId = BL.Id
+JOIN Libraries Lib ON Lib.Id = BL.LibraryId
+JOIN Books B ON B.Id = BL.BookId;
+
+--exercise6
+--Display all the librarians that are not on holiday and are hired more than 5 years ago.
+SELECT Name 
+FROM Librarians
+WHERE IsOnHoliday = 0 AND HireDate < GETDATE() - YEAR(5)
+
+--exercise7
+--Create a stored procedure that displays, based on the book title, the number of libraries in which the book is found. Call the procedure with three different book titles.
+GO
+CREATE PROCEDURE CountLibraries(
+    @BookTitle VARCHAR(50)
+)
+AS
+BEGIN
+SELECT COUNT(LibraryId) AS NumberOfLibraries
+FROM Books_Libraries
+JOIN Books ON Books.Id = Books_Libraries.BookId
+WHERE @BookTitle = Books.Title
+END
+GO
+
+exec
+CountLibraries'Prietena mea geniala';
+exec
+CountLibraries'Nascut in afara legii';
+exec
+CountLibraries'Ciuleandra';
```

Comment:
Nice. Maybe the naming should be a bit more specific like "CountLibrariesBasedOnBookTitle" or something similar?